crates/fdev/src/network_metrics_server.rs

```diff
@@ -127,7 +130,30 @@ async fn push_interface(ws: WebSocket, state: Arc<ServerState>) -> anyhow::Resul
                         break;
                     }
                     Ok(change) => {
-                        if let Err(err) = state.save_record(change) {
+                        if let Err(err) = state.save_record(ChangesWrapper::PeerChange(change)) {
+                            tracing::error!(error = %err, "Failed saving report");
+                            tx.send(Message::Binary(ControllerResponse::into_fbs_bytes(Err(
+                                format!("{err}"),
+                            ))))
+                            .await?;
+                        }
+                    }
+                    Err(decoding_error) => {
+                        tracing::error!(error = %decoding_error, "Failed to decode message");
+                        tx.send(Message::Binary(ControllerResponse::into_fbs_bytes(Err(
+                            format!("{decoding_error}"),
+                        ))))
+                        .await?;
+                    }
+                }
+                match ContractChange::try_decode_fbs(&msg) {
+                    Ok(ContractChange::PutFailure(err)) => {
+                        // FIXME: handle put failure
+                        tracing::error!(error = "Failed to put contract");
+                    }
```

Comment:
It seems that we do not really handle `PutFailure` here.

Reply:
Yes that's not done yet, but is ok, we can come back to this later if we need to.